### PR TITLE
Unify LegacyFlgDestination with CCTDestination.

### PR DESCRIPTION
**Pull request by rlazo**

### CODE REVIEW

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java [hunk 1]
```diff
@@ -17,22 +17,137 @@
...
+   * Creates a new object by parsing a {@code byte[]} generated by {@link asByteArray}.
+   *
+   * <p><strong>Important:</strong> Changes in the encoding should be backwards compatible to ensure
+   * any data persisted from a previous version can be read successfully.
+   */
+  @NonNull
+  public static CCTDestination fromByteArray(@NonNull byte[] a) {
+    String buffer;
+    try {
+      buffer = new String(a, "UTF-8");
```
Comment (vkryachko) [inline transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java]: I think if you just use `Charset.forName("UTF-8")` you won't have to deal with checked exceptions. Here and throughout.

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java [hunk 2]
```diff
@@ -17,22 +17,137 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.google.android.datatransport.runtime.Destination;
+import java.io.UnsupportedEncodingException;
+import java.util.regex.Pattern;
 
 public final class CCTDestination implements Destination {
   static final String DESTINATION_NAME = "cct";
-  public static final CCTDestination INSTANCE = new CCTDestination();
```
[the remark this answers is not in the document]
Comment (rlazo) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java]: Done.

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java [hunk 3]
```diff
@@ -17,22 +17,137 @@
...
+  static final String LEGACY_DESTINATION_NAME = "lflg";
```
[the remark this answers is not in the document]
Comment (rlazo) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java]: Done.

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java [hunk 4]
```diff
@@ -17,22 +17,137 @@
...
 
-  private CCTDestination() {}
+  private static final String DEFAULT_API_KEY =
+      StringMerger.mergeStrings("AzSCki82AwsLzKd5O8zo", "IayckHiZRO1EFl1aGoK");
+  private static final String EXTRAS_VERSION_MARKER = "1$";
+  private static final String EXTRAS_DELIMITER = "\\";
+
+  public static final CCTDestination DEFAULT_INSTANCE =
+      new CCTDestination(DESTINATION_NAME, null, null);
```
[the remark this answers is not in the document]
Comment (rlazo) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java]: Done.

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java [hunk 5]
```diff
@@ -17,22 +17,137 @@
...
+  @Nullable
+  public byte[] asByteArray() {
+    if (apiKey == null && endPoint == null) {
+      return null;
+    }
+    String buffer =
+        String.format(
+            "%s%s%s%s%s%s",
+            EXTRAS_VERSION_MARKER,
+            destinationName,
```
[the remark this answers is not in the document]
Comment (rlazo) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java]: done

**Pull request merged**